src/coreclr/vm/amd64/AsmHelpers.asm

```diff
@@ -698,5 +698,240 @@ NESTED_END OnCallCountThresholdReachedStub, _TEXT
+
+        mov             r10, [rbp + 8] ; Get address that we are going to eventually return to
+
+        add             r10, 5 ; Skip the ret opcode and stack adjustment
+        call            r10 ; Call the core of the function with the new larger stack
+
+        mov             rsp, rbx ; Bring back the old stack pointer
+
+        mov             r10, [rbp - 020h] ; Pull the old stack limit
+        mov             gs:[10h], r10 ; Change stack limit to new value
+        mov             r10, [rbp - 018h] ; Pull the old stack base
+        mov             gs:[8h], r10 ; Change stack base to new value
+
+        ; Save return value registers
+        mov             [rsp + 70h], rax
+        movdqa          [rsp + 20h], xmm0
+        
+        call            End_More_Thread_Bookeeping
+
+        ; Restore return value registers
+        mov             rax, [rsp + 70h]
+        movdqa          xmm0, [rsp + 20h]
+
+
+
+        ; Restore registers
+        mov             rbx, [rbp - 48h]
+        mov             rbp, [rbp - 50h]
+        add rsp, 0e8h
+        ret
+NESTED_END _more_stack, _TEXT
+
+
+; This transitions into a green thread, assuming that there are no stack arguments
+; and the first argument (rcx) is the address of the first function to use in the 
+; green thread
+NESTED_ENTRY GreenThread_StartThreadHelper2, _TEXT
+    alloc_stack     28h
+    END_PROLOGUE
+        mov eax, 0
+        call _more_stack
+        add rsp, 28h
+        ret
+        jmp rcx
+NESTED_END GreenThread_StartThreadHelper2, _TEXT
+
+NESTED_ENTRY GreenThread_StartThreadHelper, _TEXT
+        PROLOG_WITH_TRANSITION_BLOCK
+        call GreenThread_StartThreadHelper2
+        EPILOG_WITH_TRANSITION_BLOCK_RETURN
+NESTED_END GreenThread_StartThreadHelper, _TEXT
+
+
+extern FirstFrameInGreenThreadCpp:proc
+NESTED_ENTRY FirstFrameInGreenThread, _TEXT
+    alloc_stack     38h ;; TODO This is overly large
+    save_reg_postrsp    rbp, 20h ; Save off the rbp and rbx in well known locations, as those need to be updated when the thread resumes
+    save_reg_postrsp    rbx, 28h
+    END_PROLOGUE
+    call FirstFrameInGreenThreadCpp
+    mov             rbx, [rsp + 28h]
+    mov             rbp, [rsp + 20h]
+    add rsp, 38h
+    ret
+NESTED_END FirstFrameInGreenThread, _TEXT
+
+; This transitions into an OS thread, assuming that there are no stack arguments
+; and the first argument (rcx) is the address of the first function to use in the 
+; green thread
+NESTED_ENTRY TransitionToOSThreadHelper, _TEXT
+    alloc_stack     28h
+    END_PROLOGUE
+        mov eax, 0
+        sub eax, 1
+        call _more_stack
+        add rsp, 28h
+        ret
+        jmp rcx
+NESTED_END TransitionToOSThreadHelper, _TEXT
+
+
+; extern "C" void YieldOutOfGreenThreadHelper(StackRange *pOSStackRange, uint8_t* osStackCurrent, uint8_t** greenThreadStackCurrent);
+; This function is supposed to hop back to the OS thread, with the original OS thread stack in place, and 
```

Comment:
The comment seems unfinished.

Reply:
It does doesn't it. :) Thank you for reading enough to find this. I'll see if I can remember the thought I had before I went and got distracted trying to put something else together.